c7n/resources/asg.py

```diff
@@ -244,7 +245,8 @@ def get_permissions(self):
                       'app-elb-target-group', 'ebs-snapshot', 'ami')]))
 
     def validate(self):
-        if self.manager.data.get('mode'):
+        execution_mode = self.manager.data.get('mode', {'type': 'pull'})['type']
+        if issubclass(c7n.policy.execution.get(execution_mode), c7n.policy.LambdaMode):
```

Comment:
The resource manager is given a context when created. That context contains the policy. The policy has a method to get the execution mode.

```suggestion
        execution_mode = self.manager.ctx.policy.get_execution_mode()
        if issubclass(execution_mode, c7n.policy.LambdaMode):
```

Reply:
Doh, saw your comment after I made a very similar change. Good call though thanks :+1: